### PR TITLE
Brief documentation about how to configure HDFS for rack awareness.

### PR DESCRIPTION
**Pull request by soenkeliebau**
# Description

Adds documentation on how to configure Hdfs to use the topology provider for rack awareness.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: docs/modules/hdfs/partials/nav.adoc [hunk 1]
```diff
@@ -10,6 +10,7 @@
 ** xref:hdfs:usage-guide/resources.adoc[]
 ** xref:hdfs:usage-guide/scaling.adoc[]
 ** xref:hdfs:usage-guide/fuse.adoc[]
+** xref:hdfs:usage-guide/rack-awareness.adoc[]
```
Comment (siegfriedweber) [inline docs/modules/hdfs/partials/nav.adoc]: Rack awareness and Pod placement are – to a certain degree – similar topics. Rack awareness is placed at the top-level of the usage guides whereas Pod placement is in the Operations sub menu. I have to admit that the distinction between Operations and non-Operations is not clear to me, so I have no opinion where Rack awareness belongs to, but these two topics should be placed together.

File: docs/modules/hdfs/pages/usage-guide/rack-awareness.adoc [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+The default write path, for example, is to put replicas of a newly created block first on a different node, but within the same rack, and the second copy on a node in a remote rack.
+In order for this to work properly, Hadoop needs to have information about the underlying infrastructure it runs on available - in a Kubernetes environment, this means obtaining information from the pods or nodes of the cluster.
+
+In order to enable gathering this information the Hadoop images contain https://github.com/stackabletech/hdfs-topology-provider on the classpath, which can be configured to read labels from Kubernetes objects.
+
+In the current version of the SDP this is not exposed as fully integrated functionality in the operator, but rather needs to be configured via config overrides.
+
+
+NOTE:
+Until the operator code has been merged, users will need to manually deploy RBAC objects to allow the Hadoop pods access to the necessary Kubernetes objects.
```
Comment (siegfriedweber) [inline docs/modules/hdfs/pages/usage-guide/rack-awareness.adoc]: `NOTE:` must not be followed by a newline if it should be shown as an admonition in the rendered HTML output.

```suggestion
NOTE: Until the operator code has been merged, users will need to manually deploy RBAC objects to allow the Hadoop pods access to the necessary Kubernetes objects.
```

File: docs/modules/hdfs/pages/usage-guide/rack-awareness.adoc [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+[source,yaml]
+----
+subjects:
+  - kind: ServiceAccount
+    name: hdfs-<clustername>-serviceaccount
+    namespace: <cluster-namespace>
+----
+
+So for an HDFS cluster named `simple-hdfs` in the `stackable` namespace, the full ClusterRoleBinding would look like this:
```
Comment (siegfriedweber) [inline docs/modules/hdfs/pages/usage-guide/rack-awareness.adoc]:
```suggestion
So for an HDFS cluster using the ServiceAccount `hdfs-serviceaccount` in the `stackable` namespace, the full ClusterRoleBinding would look like this:
```

File: docs/modules/hdfs/pages/usage-guide/rack-awareness.adoc [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+kind: HdfsCluster
+metadata:
+  name: simple-hdfs
+spec:
+  nameNodes:
+    config:
+    configOverrides:
+      core-site.xml:
+        HADOOP_CLASSPATH: "/stackable/hadoop/share/hadoop/tools/lib/topology-provider-0.1.0.jar"
+        net.topology.node.switch.mapping.impl: tech.stackable.hadoop.StackableTopologyProvider
```
Comment (siegfriedweber) [inline docs/modules/hdfs/pages/usage-guide/rack-awareness.adoc]: `HADOOP_CLASSPATH` must be set in `envOverrides`:

```suggestion
    configOverrides:
      core-site.xml:
        net.topology.node.switch.mapping.impl: tech.stackable.hadoop.StackableTopologyProvider
    envOverrides:
      HADOOP_CLASSPATH: "/stackable/hadoop/share/hadoop/tools/lib/topology-provider-0.1.0.jar"
```

File: docs/modules/hdfs/pages/usage-guide/rack-awareness.adoc [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+[source,yaml]
+----
+apiVersion: hdfs.stackable.tech/v1alpha1
+kind: HdfsCluster
+metadata:
+  name: simple-hdfs
+spec:
+  nameNodes:
+    config:
```
Comment (siegfriedweber) [inline docs/modules/hdfs/pages/usage-guide/rack-awareness.adoc]: The empty `config` section can be removed:

```suggestion
```

**Review by siegfriedweber — CHANGES_REQUESTED** [review]

File: docs/modules/hdfs/partials/nav.adoc (hunk 1 above)
Comment (soenkeliebau) [reply docs/modules/hdfs/partials/nav.adoc]: moved it

**Review by siegfriedweber — APPROVED** [review]
LGTM

**Pull request merged**